tsdb/head.go

```diff
@@ -1072,6 +1077,15 @@ func (h *Head) gc() int64 {
 	// Remove deleted series IDs from the postings lists.
 	h.postings.Delete(deleted)
 
+	// Remove tombstones referring to the deleted series.
+	refs := make([]uint64, 0, len(deleted))
+	for ref := range deleted {
+		refs = append(refs, ref)
+	}
```

Comment:
Apologies, I was not aware that `deleted` was a map :)
Let's pass `map[uint64]struct{}` in `DeleteTombstones` to avoid this conversion.